src/main/java/soc/game/SOCPlayer.java

```diff
@@ -2916,7 +2916,7 @@ else if (boardHasVillages)
             if (edge == -9)
                 continue;
             SOCRoutePiece pp = getRoadOrShip(edge);
-            if ((pp == null) || ! (pp instanceof SOCShip))
+            if (! (pp instanceof SOCShip))
```

Comment:
Are you sure that's never null?

Reply:
The null check is redundant. instanceof will always return false if the object is null.

`public class NullInstanceOf {

	public static void main(String args[]) {
		Object o = null;
		
		if (o == null) {
			System.out.println("o is null!");
		}
		
		if (o instanceof Object) {
			System.out.println("o is an object");
		} else {
			System.out.println("o is not an object");
		}
	}

}`

instanceof $ java NullInstanceOf
o is null!
o is not an object